app/src/main/java/com/allan/boardbuddies/activities/CanvasActivity.java

```diff
@@ -47,6 +57,18 @@ public void onGlobalLayout() {
                 }
             });
         }
+        bottomNavigationView = findViewById(R.id.canvas_bottom_nav);
+        bottomNavigationView.setOnItemSelectedListener(this);
+        getSupportFragmentManager().setFragmentResultListener("requestKey", this, new FragmentResultListener() {
```

Comment:
You should use a specific request key, since this listens to everything from the fragment manager, and indicate the fragment and/or the result - e.g. "TEXT_DIALOG_FRAGMENT_ADD_TEXT_REQUEST_KEY"

Reply:
Would recommend making this a global constant^^